include/onnxruntime/core/providers/tensorrt/tensorrt_provider_options.h

```diff
@@ -64,10 +64,20 @@ struct OrtTensorRTProviderOptionsV2 {
    *       - if "trt_engine_cache_path" is "" -> the engine cache will be saved to "./context_model_dir"
    *       - if "trt_engine_cache_path" is "engine_dir" -> the engine cache will be saved to "./context_model_dir/engine_dir"
    *
+   * 3. In the case of building weightless engines, the same security reasons as listed in 1) apply to the
+   *    "onnx_model_filename" node attribute of EP context node, which contains a filename of the ONNX model with the
+   *    weights needed for the refit process. User can specify a folder path relative to the current working
+   *    directory by means of the "trt_onnx_model_folder_path" option.
+   *
    */
   int trt_dump_ep_context_model{0};               // Dump EP context node model
   const char* trt_ep_context_file_path{nullptr};  // Specify file name to dump EP context node model. Can be a path or a file name or a file name with path.
   int trt_ep_context_embed_mode{0};               // Specify EP context embed mode. Default 0 = context is engine cache path, 1 = context is engine binary data
+  int trt_weightless_engine_enable{0};            // Enable weightless engine build. Default 0 = false,
```

Comment:
Is the user able to serialize the full engine after refitting? After reading through this MR it doesn't seem like that's possible.

Reply:
Not for now, as discussed with Max, we'd like the user being able to always choose the same session options, i.e. ultimately choosing between the `model_ctx.onnx` (if present) and `model.onnx` (using it first once in the absence of the former). The serialized engine with refitted weights would most likely introduce a yet another EP option to the flow.